tokio/src/macros/select.rs

```diff
@@ -430,7 +449,8 @@ macro_rules! select {
         // This module is defined within a scope and should not leak out of this
         // macro.
         mod util {
-            // Generate an enum with one variant per select branch
+            // Generate an enum with one variant per select branch, including
+            // the branches which cfg is false.
             $crate::select_priv_declare_output_enum!( ( $($count)* ) );
```

Comment:
I'm pretty sure you can get it to not generate the branch at all if you modify the `select_priv_declare_output_enum` macro to generate output like this:
```Rust
pub(super) enum Out<_0, #[cfg(my_condition)] _1, _2> {
    _0(_0),
    #[cfg(my_condition)]
    _1(_1),
    _2(_2),
    Disabled,
}

pub(super) type Mask = u8;
```
Though you still can't have it "compress" the range of integers to the actual number of enabled ones, so you'll probably need to add an `_ => continue` branch to the match in the loop.

Reply:
One issue is modify `select_priv_declare_output_enum`, but it's not really the issue I am concern with. A bigger issue is `let (_, #[cfg(...)] _, ..) = ...;` is not possible.